llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -5514,15 +5516,25 @@ InstructionCost AArch64TTIImpl::getShuffleCost(
         else
           NMask.push_back(MaskElt % LTNumElts);
       }
+      // Check if we have already generated this sub-shuffle, which means we
+      // will have already generated the output. For example a <16 x i32> splat
+      // will be the same sub-splat 4 times, which only needs to be generated
+      // once and reused.
+      auto Result =
+          PreviousCosts.insert({std::make_tuple(Source1, Source2, NMask), 0});
+      if (!Result.second)
+        continue;
```

Comment:
It wasn't immediately clear that `Result.second` isn't the map's entry's value and is actually whether the entry was inserted or not. Could you explain that a little in the comment?